src/launchpad/size/insights/apple/image_optimization.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+import io
+import logging
+
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Iterable, List, Sequence
+
+import pillow_heif  # type: ignore
+
+from PIL import Image
+
+from launchpad.size.insights.insight import Insight, InsightsInput
+from launchpad.size.models.apple import (
+    ImageOptimizationInsightResult,
+    OptimizableImageFile,
+)
+from launchpad.size.models.common import FileInfo
+from launchpad.utils.logging import get_logger
+
+pillow_heif.register_heif_opener()  # type: ignore
+
+# Silence noisy loggers
+for noisy in ("PIL", "pillow_heif"):
+    logging.getLogger(noisy).setLevel(logging.WARNING)
+
+logger = get_logger(__name__)
+
+
+@dataclass(slots=True)
+class _OptimizationResult:
+    savings: int
+    optimized_size: int
+
+
+class ImageOptimizationInsight(Insight[ImageOptimizationInsightResult]):
+    """Analyse image optimisation opportunities in iOS apps."""
+
+    OPTIMIZABLE_FORMATS = {"png", "jpg", "jpeg", "heif", "heic"}
+    MIN_SAVINGS_THRESHOLD = 4096
+    TARGET_JPEG_QUALITY = 85
+    TARGET_HEIC_QUALITY = 85
+    _MAX_WORKERS = 8
+
+    def generate(self, input: InsightsInput) -> ImageOptimizationInsightResult | None:  # noqa: D401
+        files = list(self._iter_optimizable_files(input.file_analysis.files))
+        if not files:
+            return None
+
+        results: List[OptimizableImageFile] = []
+        with ThreadPoolExecutor(max_workers=min(self._MAX_WORKERS, len(files))) as executor:
```

Comment:
This was slower than I hoped so I threw it in a threadpool. I'll monitor the performance (e.g. peak memory) of this.